sycl/test-e2e/Regression/reduction_64bit_atomic64.cpp

```diff
@@ -0,0 +1,59 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+//
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+// Tests that a previously known case for reduction doesn't cause a requirement
+// for atomic64.
+// TODO: When aspect requirements are added to testing, this test could be set
+//       to require that atomic64 is NOT supported, to limit how frequently the
+//       test is run. However, it should work on devices that support atomic64
+//       as well.
```

Comment:
I'd suggest adding runtime aspect check for that here in this PR with the expectation that we'll uplift it to `// REQUIRES` once the support is added to the `llvm.lit.cfg.py`.

I also think that a slightly better option would be to "unit-test" the dynamic strategy selector once it's implemented (i.e., don't even run the actual reduction kernel).

Reply:
> I'd suggest adding runtime aspect check for that here in this PR with the expectation that we'll uplift it to `// REQUIRES` once the support is added to the `llvm.lit.cfg.py`.

We could do a runtime check if you prefer. The test is still valid, so this TODO is more of a suggestion than a requirement really.

> I also think that a slightly better option would be to "unit-test" the dynamic strategy selector once it's implemented (i.e., don't even run the actual reduction kernel).

I agree that would be better, but sadly the unittests don't compile the kernels, so we would have no way of knowing if the corresponding kernels had any requirements. We could set the requirements ourselves, but that really defeats the purpose of the tests.